components/amount-buttons.jsx

```diff
@@ -83,7 +83,9 @@ var AmountButtons = React.createClass({
       userInputting: userInputting,
       valid: true
     });
-    this.props.onChange(this.props.name, this, amount);
+    this.props.onChange(this.props.name, this, {
```

Comment:
The onChange now uses a generic function, because frequency has two props.

This means amount needs to pass in amount as a prop, because it *could* have multiple values.